codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2FrameReader.java

```diff
@@ -478,6 +485,13 @@ public void processFragment(boolean endOfHeaders, ByteBuf fragment,
         // Process the initial fragment, invoking the listener's callback if end of headers.
         final ByteBuf fragment = payload.readSlice(lengthWithoutTrailingPadding(payload.readableBytes(), padding));
         headersContinuation.processFragment(flags.endOfHeaders(), fragment, listener);
+        resetHeadersContinuationIfEnd(flags.endOfHeaders());
+    }
+
+    private void resetHeadersContinuationIfEnd(boolean endOfHeaders) {
+        if (endOfHeaders) {
```

Comment:
`headersContinuation` must be `close`d before it is set to `null`.

Consider created a utility method and sharing with `close()` in this class.

```java
@Override
public void close() {
  closeHeadersContinuation();
}

private void closeHeadersContinuation() {
  if (headersContinuation != null) {
    headersContinuation.close();
    headersContinuation = null;
  }
}

private void resetHeadersContinuationIfEnd(boolean endOfHeaders) {
  if (endOfHeaders) {
    closeHeadersContinuation();
  }
}
```